File: services/ocr-service/config/environment.py
import os
from dotenv import load_dotenv
from .types import EnvironmentConfig

# Load environment variables from .env file
load_dotenv()

def get_environment_config() -> EnvironmentConfig:
    env = os.getenv('ENV', 'development')
    node_env = os.getenv('NODE_ENV', env)
    port = int(os.getenv('PORT', '8001'))

    return EnvironmentConfig(
        env=env,
        node_env=node_env,
        port=port,
        debug=os.getenv('DEBUG', 'false').lower() == 'true',
        is_production=env == 'production',
        is_development=env == 'development'
    )

File: packages/api/src/config/environment.ts
import type { EnvironmentConfig } from './types'

const nodeEnv = (process.env.NODE_ENV || 'development') as EnvironmentConfig['nodeEnv']

export const environmentConfig: EnvironmentConfig = {
  nodeEnv,
  isProduction: nodeEnv === 'production',
  isDevelopment: nodeEnv === 'development'
}